CoinDock-api/app/Models/V1/User.php

```diff
@@ -100,4 +102,99 @@ public function signUp()
+        $balanceInUsd = $this->wallets
+            ->mapToGroups(function ($wallet) {
+                return ['balance' => $wallet->balance];
+            })
+            ->map(function ($e) {
+                return $e->sum();
+            });
+        return [
+            'coin_name' => $primaryCurrency,
+            'balance'  => $primaryBalancePath['rate'] * $balanceInUsd['balance']
+        ];
+    }
+
+
+    public function topPerformer():array
+    {
+        $walletCoinIds = $this->wallets()->pluck('coin_id');
+        $coins = Coin::select(['coin_id', 'name'])->whereIn('id', $walletCoinIds)->get();
+        $baseUrl = config('coin.coin.api_url');
+        $currencyURL = $baseUrl . config('coin.coin.top_performer');
+        $topPerformerBal = PHP_INT_MIN;
+        $coinName = Null;
+        $shortName = Null;
+        foreach ($coins as $coin) {
+            $currency = str_replace('{id}', $coin->coin_id, $currencyURL);
+            $primaryBalancePath = Http::withHeaders(['X-CoinAPI-Key' => config('coin.coin.api_key')])->get($currency);
+            if ($primaryBalancePath['rate'] > $topPerformerBal) {
+                $topPerformerBal = $primaryBalancePath['rate'];
+                $shortName = $primaryBalancePath['asset_id_base'];
+                $coinName  = Coin::whereCoinId($shortName)->first()?->name;
+            }
+        }
+            return[
+                'balance' => $topPerformerBal,
+                'coin_name' => $coinName,
+                'coin_id' =>  $shortName
+            ];
+    }
+
+    public function lowPerformer(): array
+    {
+        $walletCoinIds = $this->wallets()->pluck('coin_id');
+        $coins = Coin::select(['coin_id', 'name'])->whereIn('id', $walletCoinIds)->get();
+            $baseUrl = config('coin.coin.api_url');
+            $currencyURL = $baseUrl . config('coin.coin.top_performer');
+            $lowPerformerBal = PHP_INT_MAX;
+            $coinName = Null;
+            $shortName= Null;
+            foreach ($coins as $coin) {
+                $currency = str_replace('{id}', $coin->coin_id, $currencyURL);
+                $primaryBalancePath = Http::withHeaders(['X-CoinAPI-Key' => config('coin.coin.api_key')])->get($currency);
+                if ($primaryBalancePath['rate'] < $lowPerformerBal) {
+                    $lowPerformerBal = $primaryBalancePath['rate'];
+                    $shortName= $primaryBalancePath['asset_id_base'];
+                    $coinName  = Coin::whereCoinId($shortName)->first()?->name;
+                }
+            }
+            return [
+                'balance' => $lowPerformerBal,
+                'coin_name' => $coinName,
+                'coin_id' => $shortName
+            ];
+        
+    }
+
+    public function wallets()
+    {
+        return $this->hasMany(Wallet::class, 'user_id', 'id');
+    }
+
+    public function settings()
+    {
```

Comment:
if the relation is hasOne() then the function name should be singular.

Reply:
changed